### PR TITLE
LibJS: Use is_unicode_surrogate more

### PR DESCRIPTION
**Pull request by nico**
And add some spec comments.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/StringPrototype.cpp [hunk 1]
```diff
@@ -110,23 +110,38 @@ static bool is_string_well_formed_unicode(Utf16View string)
...
     VERIFY(position < string.length_in_code_units());
 
+    // 3. Let first be the code unit at index position within string.
     auto first = string.code_unit_at(position);
+
+    // 4. Let cp be the code point whose numeric value is that of first.
     auto code_point = static_cast<u32>(first);
 
+    // 5. If first is not a leading surrogate or trailing surrogate, then
     if (!Utf16View::is_high_surrogate(first) && !Utf16View::is_low_surrogate(first))
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/StringPrototype.cpp]: Please add curly braces here (https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#wrong-27), same below

File: Userland/Libraries/LibJS/Runtime/JSONObject.cpp [hunk 2]
```diff
@@ -377,14 +382,22 @@ DeprecatedString JSONObject::quote_json_string(DeprecatedString string)
             builder.append("\\\\"sv);
             break;
         default:
+            // b. Else if C has a numeric value less than 0x0020 (SPACE), or if C has the same numeric value as a leading surrogate or trailing surrogate, then
             if (code_point < 0x20 || Utf16View::is_high_surrogate(code_point) || Utf16View::is_low_surrogate(code_point)) {
+                // i. Let unit be the code unit whose numeric value is that of C.
+                // ii. Set product to the string-concatenation of product and UnicodeEscape(unit).
                 builder.appendff("\\u{:04x}", code_point);
+            // c. Else,
             } else {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/JSONObject.cpp:391]: we've settled on using this style:

```cpp
}
// c. Else,
else {
```

File: Userland/Libraries/LibJS/Runtime/JSONObject.cpp [hunk 3]
```diff
@@ -383,7 +383,7 @@ DeprecatedString JSONObject::quote_json_string(DeprecatedString string)
             break;
```
Comment (trflynn89) [inline Userland/Libraries/LibJS/Runtime/JSONObject.cpp:383]: Commit title should just be "LibJS: " unless there's other cases you're planning on adding to this commit

**Comment by linusg** [comment]
Thanks, better spec comment coverage is always appreciated :)

**Pull request merged**